### PR TITLE
feat(expo-config-plugin): create config plugin for react-native-performance

### PR DESCRIPTION
**Pull request by friyiajr**
**Motivation**

Expo is becoming an increasingly popular way of building React-Native projects. It is important that we support their ecosystem in order to keep this library relevant in the RN Community.

**NOTE:** Reverted the [import change](https://github.com/Shopify/react-native-performance/pull/118/files) found here because it seemed to be breaking builds in Vanilla RN and Expo Projects. Please double check this fix works on your end @ElviraBurchik. 

## Description

Fixes N/A

This Pull request adds a plugin folder to the library that has the code for the Expo Config-Plugin. No other code was touched. The purpose of the Config plugin is just to update the code on the library consumer's end

### Test plan

1. Checkout this branch
2. At the root of the project run `yarn` to install all dependancies
3. Run `yarn build` to build the config plugin locally
4. `cd` into the `ExpoFixture` and run `yarn` again to install it's dependancies
5. Run `npx expo prebuild` to do all native code setup
6. After this run the app on both platforms making sure that you can see the logs in Flipper

For iOS:

```bash
npx expo run:ios
```

For Android:

```bash
npx expo run:android
```

## Checklist

- [ ] I have added a decision record entry, PR includes changes to monorepo setup that may require explanation

### CODE REVIEW

File: packages/react-native-performance/plugin/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+const MATCH_DEPENDANCIES = /dependencies {/;
+
+const withRequiredIOSSetup: ConfigPlugin = expoConfig => {
+  const importStatement = '#import <ReactNativePerformance/ReactNativePerformance.h>';
+  const newLine = '  [ReactNativePerformance onAppStarted];\n';
+
+  return withAppDelegate(expoConfig, iosConfig => {
+    if (!['objc', 'objcpp'].includes(iosConfig.modResults.language)) {
+      throw new Error(
+        "react-native-google-cast config plugin does not support AppDelegate' that aren't Objective-C(++) yet.",
```
Comment (sdowding-rbi) [inline packages/react-native-performance/plugin/src/index.ts:24]:
```suggestion
        "react-native-performance config plugin does not support AppDelegate' that aren't Objective-C(++) yet.",
```

copy pasta error 🍝

Comment (friyiajr) [reply packages/react-native-performance/plugin/src/index.ts:24]: Thanks for pointing this out. I'll rename both errors 👍

Comment (sdowding-rbi) [reply packages/react-native-performance/plugin/src/index.ts]: Thank you for your work on this, I really appreciate it.

Comment (friyiajr) [reply packages/react-native-performance/plugin/src/index.ts]: No problem! I am currently working with the team in charge of this to try and take the next steps to get it merged.

**Comment by gstcarv** [comment]
any plans to merge it?

**Comment by markrickert** [comment]
@friyiajr I was able to get this config plugin working with expo sdk 50 (which uses kotlin files now in their prebuild)

```ts
const withRequiredAndroidMainApplicationSetup: ConfigPlugin = (expoConfig) => {
  const importStatement =
    'com.shopify.reactnativeperformance.ReactNativePerformance';
  return withMainApplication(expoConfig, async (androidConfig) => {
    const isJava = androidConfig.modResults.language === 'java';
    const newLine = `    ReactNativePerformance.onAppStarted()${
      isJava ? ';' : ''
    }\n`;

    const srcWithNewImport = addImports(
      androidConfig.modResults.contents,
      [importStatement],
      isJava
    );

    androidConfig.modResults.contents = mergeContents({
      tag: 'react-native-performance-onCreate-android',
      src: srcWithNewImport,
      newSrc: newLine,
      anchor: MATCH_ON_CREATE,
      offset: 0,
      comment: '//',
    }).contents;

    return androidConfig;
  });
};
```

**Comment by GrinZero** [comment]
Will this PR be restored?

**Pull request closed without merging**